src/App/Pages/Accounts/TwoFactorPageViewModel.cs

```diff
@@ -245,28 +267,32 @@ private async Task DuoFramelessAuthenticateAsync()
                 return;
             }
 
-            string response = null;
-            if (authResult != null && authResult.Properties.TryGetValue("code", out var resultData))
+            await _deviceActionService.HideLoadingAsync();
+            if (authResult == null || authResult.Properties == null)
             {
-                response = Uri.UnescapeDataString(resultData);
+                throw new InvalidOperationException("Duo authentication error: Could not get result from authentication");
             }
-            if (!string.IsNullOrWhiteSpace(response))
+
+            if (authResult.Properties.TryGetValue("error", out var resultError))
             {
-                Token = response;
-                await SubmitAsync(false);
+                _logger.Error(resultError);
+                await _platformUtilsService.ShowDialogAsync(AppResources.AnErrorHasOccurred, AppResources.Ok);
                 return;
             }
 
-            await _deviceActionService.HideLoadingAsync();
-            if (authResult != null && authResult.Properties.TryGetValue("error", out var resultError))
+            string code = null;
+            if (authResult.Properties.TryGetValue("code", out var resultData))
```

Comment:
❓ Can `Properties` be `null` here?

Reply:
this check is already been done on ln 271. Since it was used twice i did the check before hand.